sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -35,12 +59,37 @@ public virtual async Task<Response<Completions>> GetCompletionsAsync(string depl
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         public virtual Response<Completions> GetCompletions(string deploymentId, string prompt, CancellationToken cancellationToken = default)
         {
-            Argument.AssertNotNullOrEmpty(deploymentId, nameof(deploymentId));
             Argument.AssertNotNullOrEmpty(prompt, nameof(prompt));
-
             CompletionsOptions completionsOptions = new CompletionsOptions();
             completionsOptions.Prompt.Add(prompt);
             return GetCompletions(deploymentId, completionsOptions, cancellationToken);
         }
+
+        internal HttpMessage CreateGetCompletionsRequest(string deploymentId, RequestContent content, RequestContext context)
```

Comment:
Do we need to also make this customization for the embeddings and streaming APIs?

We should avoid having a client that have methods that don't work simply due to which constructor you used.

Reply:
Yes, I will add those as well.